memcache/memcache.go

```diff
@@ -463,10 +473,14 @@ func parseGetResponse(r *bufio.Reader, cb func(*Item)) error {
 		if err != nil {
 			return err
 		}
-		it.Value, err = ioutil.ReadAll(io.LimitReader(r, int64(size)+2))
+		buf := bufferPool.Get().(*bytes.Buffer)
+		_, err = buf.ReadFrom(io.LimitReader(r, int64(size)+2))
 		if err != nil {
+			freeBuffer(buf)
 			return err
 		}
+		it.Value = buf.Bytes()
+		freeBuffer(buf)
```

Comment:
this isn't safe. You're still using it.Value but giving it back to the pool where somebody else can populate it.